### DISCUSSION TITLE
laser frequency error

### DISCUSSION THREAD

**Discussion by dpkhatua** [epochpic/epoch#569] [General]
Hi all,
I tried one simulation only with a laser pulse of 20 fs at 1090nm central wavelength. I extracted the |E| of the laser at the center point of the computation window in the time domain and did the Fourier transform to see all the frequencies. I found that other than the fundamental frequency (550 THz), higher harmonics were also generated. I am attaching the input.deck and Fourier transform data here. Please take a look at it and suggest if any special attention has to be given.

![lasaer_frequency](https://github.com/Warwick-Plasma/epoch/assets/63596383/e423c79d-c2be-4c00-8621-670cbb370499)
![E_laser_time](https://github.com/Warwick-Plasma/epoch/assets/63596383/3d91bd72-049d-45a4-85ef-84c49343dc6f)
[input.txt](https://github.com/Warwick-Plasma/epoch/files/13125700/input.txt)

**Comment by Status-Mirror** [comment]
Hey @Khatua03,

Just by counting peaks in your figure, I can see that the laser oscillates with a frequency of around 300 THz, which doesn't have a peak on your Fourier spectrum. Are you sure this Fourier spectrum is correct? Maybe you shouldn't be including data before the pulse arrives in your spectrum?

Cheers,
Stuart

**Reply by dpkhatua** [reply]
Hi Stuart,
Thanks for the quick response. Yes, you are right. The laser frequency is ~275 THz. The plot I showed here corresponds to the amplitude of the electric field, which is always positive, and the Fourier transform code sees the frequency as a double of it. I am considering all the data points even before the arrival of the laser. In this case, I am just wondering why the higher harmonics are appearing here. If I consider the y component of the electric field, only the fundamental frequency, i.e., 275 THz, can be seen in the fft and no higher harmonics. Is this a numerical error?

Thanks.
Durga Prasad

**Reply by Status-Mirror** [reply]
It's possible the higher harmonics are an artefact of the field-solver used by EPOCH. The boundary treatment may also be introducing this noise, you could try increasing `y_min` and `y_max`? Also, the propagation direction changes with $y$ for Gaussian beams, so $E_y$ will not fully characterise your laser - maybe this is the source of the higher harmonics?

I don't know enough about the Fourier description of the beam to comment further unfortunately.

**Comment by amarsarari** [comment]
Dear sir can you provide matlab sripts to see above plots
